### PR TITLE
Add separate error handler for 405(Method not allowed) errors

### LINKED ISSUE #26375: Airflow Webserver returns incorrect HTTP Error Response for custom REST API endpoints
### Apache Airflow version

Other Airflow 2 version

### What happened

We are using Airflow 2.3.1 Version. Apart from Airflow provided REST endpoints, we are also using the airflow webserver to host our own application REST API endpoints. We are doing this by loading our own blueprints and registering Flask Blueprint routes within the airflow plugin. 

Issue: Our Custom REST API endpoints are returning incorrect HTTP Error response code of 404 when 405 is expected (Invoke the REST API endpoint with an incorrect HTTP method, say POST instead of PUT) . This was working in airflow 1.x but is giving an issue with airflow 2.x

Here is a sample airflow plugin code . If the '/sample-app/v1' API below is invoked with POST method, I would expect a 405 response. However, it returns a 404. 

I tried registering a blueprint error handler for 405 inside the plugin, but that did not work.

```
test_bp = flask.Blueprint('test_plugin', __name__)

@test_bp.route(
    '/sample-app/v1/tags/<tag>', methods=['PUT'])
def initialize_deployment(tag):
    """
    Initialize the deployment of the metadata tag

    :rtype: flask.Response
    """
    return 'Hello, World'

class TestPlugin(plugins_manager.AirflowPlugin):
    name = 'test_plugin'
    flask_blueprints = [test_bp]
```

### What you think should happen instead

Correct HTTP Error response code should be returned.

### How to reproduce

Issue the following curl request after loading the plugin - 

curl -X POST "http://localhost:8080/sample-app/v1/tags/abcd" -d '' 

The response will be 404 instead of 405.

### Operating System

Ubuntu

### Versions of Apache Airflow Providers

_No response_

### Deployment

Docker-Compose

### Deployment details

_No response_

### Anything else

_No response_

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

### PR DESCRIPTION
**Pull request by dakshin-k**
Previously, the webserver was using the same error handler for both 404 and 405 errors, and returning a 'Not Found' response for both cases. Hence the response status code was always 404, even when a 405(Method not allowed) was raised internally.

This PR creates a separate error handler for 405 errors, so it will return an appropriate response and status code.

closes: #26375

### Results of manual testing

- Returns a JSON response when hitting the REST APIs
```bash
dakshin@bluepenguin:~/projects/airflow$ curl -X PATCH -D - http://localhost:28080/api/v1/version
HTTP/1.1 405 METHOD NOT ALLOWED
Server: gunicorn
Date: Wed, 05 Oct 2022 08:35:30 GMT
Connection: close
Content-Type: application/problem+json
Content-Length: 142
X-Robots-Tag: noindex, nofollow
Set-Cookie: session=e2b057f0-12bb-4275-8d7c-ec10f86f291f.T49QxckJWF1FK5zE-nOHZMkMVg8; Expires=Fri, 04 Nov 2022 08:35:30 GMT; HttpOnly; Path=/; SameSite=Lax

{
  "detail": "The method is not allowed for the requested URL.",
  "status": 405,
  "title": "Method Not Allowed",
  "type": "about:blank"
}
```
- Returns a HTML response for other URLs
```bash
dakshin@bluepenguin:~/projects/airflow$ curl -X PATCH -D - http://localhost:28080/home
HTTP/1.1 405 METHOD NOT ALLOWED
Server: Werkzeug/2.2.2 Python/3.7.14
Date: Wed, 05 Oct 2022 08:42:30 GMT
Content-Type: text/html; charset=utf-8
Content-Length: 475
X-Robots-Tag: noindex, nofollow
Set-Cookie: session=1b653de3-8dfa-477f-a4b5-72bbef8bfaa9.vuiHP-jyaAoquKD_OAd5DEdQVmQ; Expires=Fri, 04 Nov 2022 08:42:30 GMT; HttpOnly; Path=/; SameSite=Lax
Connection: close



<!DOCTYPE html>
<html lang="en">
  <head>
    <title>Airflow 405</title>
    <link rel="icon" type="image/png" href="/static/pin_32.png">
  </head>
  <body>
    <div style="font-family: verdana; text-align: center; margin-top: 200px;">
      <img src="/static/pin_100.png" width="50px" alt="pin-logo" />
      <h1>Airflow 405</h1>
      <p>Received an invalid request.</p>
      <a href="/">Return to the main page</a>
      <p>bdb986291012</p>
    </div>
  </body>
</html>
```

### CODE REVIEW

File: airflow/www/views.py [hunk 1]
```diff
@@ -491,6 +491,19 @@ def not_found(error):
     )
 
 
+def method_not_allowed(error):
+    """Show Method Not Allowed on screen for any error in the Webserver"""
+    return (
+        render_template(
+            'airflow/method_not_allowed.html',
```
Comment (uranusjr) [inline airflow/www/views.py]: I feel we could merge this and the 404 template instead of duplicating most of the content.

Comment (potiuk) [reply airflow/www/views.py]: Agree. we could easily templatise the html template to rednder differently in  both cases.

Comment (dakshin-k) [reply airflow/www/views.py]: Thanks both. I updated the PR based on this comment, could you please take a look again?

**Review by potiuk — APPROVED** [review]

**Comment by potiuk** [comment]
Static checks are failing - I suggest to install pre-commit locally (it will even fix things for you and running `breeze static-checks --last-commit` should run pre-commit on latest commit.

BTW. I will not be available for next two weeks (vacations) so I hope @uranusjr or someone else will merge it before :)

**Comment by dakshin-k** [comment]
Fixed the static check issues, need approval to run the workflow again.

@uranusjr, do take a look if you can :smile:

**Review by ephraimbuddy — APPROVED** [review]

**Review by uranusjr — APPROVED** [review]

**Pull request merged**